File: src/main/java/com/zerone/core/Scene.java
package com.zerone.core;

public abstract class Scene {

    protected final World world = new World();

    public abstract void onCreate();

    public abstract void onResume();

    public void update() {
        world.update();
    }

    public void draw() {

    }

    public abstract void onPause();

    public abstract void onDestroy();
}

File: src/main/kotlin/com/zerone/core/Scene.kt
package com.zerone.core

abstract class Scene {

//    protected val world = World()

    abstract fun onCreate()

    abstract fun onResume()

    open fun update() {
//        world.update()
    }

    open fun draw() {

    }

    abstract fun onPause()

    abstract fun onDestroy()
}
